munch/__init__.py

```diff
@@ -259,6 +263,42 @@ def __repr__(self):
             type(self).__name__, self.__undefined__, dict.__repr__(self))
 
 
+class DefaultFactoryMunch(defaultdict, Munch):
+    """ A Munch that calls a user-specified function to generate values for
+        missing keys like collections.defaultdict.
+
+        >>> b = DefaultFactoryMunch(list, {'hello': 'world!'})
+        >>> b.hello
+        'world!'
+        >>> b.foo
+        []
+        >>> b.bar.append('hello')
+        >>> b.bar
+        ['hello']
+    """
+    def __init__(self, *args, **kwargs):
+        if not args:
+            def default():
+                return None
+
+            args = [default]
+
+        super(DefaultFactoryMunch, self).__init__(*args, **kwargs)
+
+    @classmethod
+    def fromDict(cls, d, default=lambda: None):
+        # pylint: disable=arguments-differ
+        return munchify(d, factory=lambda d_: cls(default, d_))
+
+    def copy(self):
+        return type(self).fromDict(self, default=self.default_factory)
+
+    def __repr__(self):
```

Comment:
Why not just use `defaultdict`'s `__repr__`?

Reply:
`defaultdict`'s `__repr__` is not reversible since it doesn't get the name of the function.

Reply:
What do you mean by reversible?

Reply:
@vmalloc like with Munch, ```b = exec(repr(DefaultFactoryMunch(list)))``` works